clippy_lints/src/methods/get_last_with_len.rs

```diff
@@ -0,0 +1,55 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::path_to_local;
+use clippy_utils::source::snippet_with_applicability;
+use rustc_ast::LitKind;
+use rustc_errors::Applicability;
+use rustc_hir::{BinOpKind, Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_middle::ty;
+use rustc_span::source_map::Spanned;
+use rustc_span::sym;
+
+use super::GET_LAST_WITH_LEN;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, recv: &Expr<'_>, arg: &Expr<'_>) {
+    if let Some(recv_local) = path_to_local(recv)
```

Comment:
This is significantly more restrictive than it used to be. Something like `x.y.get(..)` or `x[0].get(..)` would have been linted before.

I'd think going back to using `SpanlessEq`, but adding a call to `Expr::can_have_side_effects` would be better.